test/unit/extension/lib/shopgate/CartExtensionPipelineTest.js

```diff
@@ -76,4 +77,23 @@ describe('CartExtensionPipeline - unit', () => {
 
     return subjectUnderTest.getCheckoutUrl().should.eventually.equal(expectedUrl)
   })
+  it('should return true when update product runs without error', function () {
+    bigCommerceCartRepositoryMock.expects('updateItems').once().returns()
+
+    return subjectUnderTest.updateProducts([{ productId: '1', quantity: 1 }]).should.eventually.equal(true)
+  })
+
+  it('should return false when update product encounters a non-breaking error', function () {
+    bigCommerceCartRepositoryMock.expects('updateItems').once().callsFake((items, notify) => {
+      notify({
```

Comment:
Make an assertion that the provided item id and quantity are the same ones that were requested. This should point you that there is an error in updateProducts call.

Reply:
I think I have met this objective by setting up a spy of the context.log.error method.

Reply:
Not entirely, all the time you have an error in your test that you're unable to see. I'm trying to help you out in finding it.
This would be true if you had no error in the test, but then still, think about it if it has happened to you during development to have an error in there and that the test seems legit, there is a chance that this happens in maintenance too which makes test unreliable. The worst are false positives.  Therefore it is worth the effort.

Reply:
revisited this and you showed me what was going on.